src/lib/server/s3.ts

```diff
@@ -10,32 +9,29 @@ import {
 import * as AWS from '@aws-sdk/client-s3';
 import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
 
-const s3client = building
-	? (null as unknown as AWS.S3)
+const s3client = new AWS.S3({
+	endpoint: S3_ENDPOINT_URL,
+	region: S3_REGION,
+	credentials: { accessKeyId: S3_KEY_ID, secretAccessKey: S3_KEY_SECRET },
+	// Handle minio, eg http://minio:9000/bucket instead of http://bucket.minio:9000
+	// Should work with any S3-compatible server, so no harm in leaving it on
+	forcePathStyle: true
+});
+
+/**
+ * To use with getSignedUrl. If PUBLIC_S3_ENDPOINT_URL is set, it will be used instead of S3_ENDPOINT_URL.
+ */
+const publicS3Client = !PUBLIC_S3_ENDPOINT_URL
+	? s3client
 	: new AWS.S3({
-			endpoint: S3_ENDPOINT_URL,
+			endpoint: PUBLIC_S3_ENDPOINT_URL,
 			region: S3_REGION,
 			credentials: { accessKeyId: S3_KEY_ID, secretAccessKey: S3_KEY_SECRET },
 			// Handle minio, eg http://minio:9000/bucket instead of http://bucket.minio:9000
 			// Should work with any S3-compatible server, so no harm in leaving it on
 			forcePathStyle: true
 	  });
 
-/**
- * To use with getSignedUrl. If PUBLIC_S3_ENDPOINT_URL is set, it will be used instead of S3_ENDPOINT_URL.
- */
-const publicS3Client =
-	building || !PUBLIC_S3_ENDPOINT_URL
-		? s3client
-		: new AWS.S3({
-				endpoint: PUBLIC_S3_ENDPOINT_URL,
-				region: S3_REGION,
-				credentials: { accessKeyId: S3_KEY_ID, secretAccessKey: S3_KEY_SECRET },
-				// Handle minio, eg http://minio:9000/bucket instead of http://bucket.minio:9000
-				// Should work with any S3-compatible server, so no harm in leaving it on
-				forcePathStyle: true
-		  });
```

Comment:
For info, `publicS3Client` uses in the case of docker compose, where the browser accesses minio through a different URL than the bebop process (which uses internal `minio` host instead of `localhost`)